aten/src/ATen/core/Tensor.h

```diff
@@ -677,6 +677,15 @@ struct CAFFE2_API WeakTensor {
 private:
   c10::weak_intrusive_ptr<TensorImpl, UndefinedTensorImpl> weak_impl_;
 };
+
+// Helper creator for Tensor clas which doesn't requires the users to pass
+// in an intrusive_ptr instead it just converts the argument passed to
+// requested intrusive_ptr type.
+template <typename T, typename... Args>
+Tensor make_tensor(Args&&... args) {
```

Comment:
Is this ever supposed to be used outside of the library? If no, can we put it in a `detail` namespace? Otherwise it might end up being confusing for people